cli/internal/graph/graph.go

```diff
@@ -76,6 +78,57 @@ func (g *CompleteGraph) GetPackageTaskVisitor(ctx gocontext.Context, visitor fun
 	}
 }
 
+// GetPipelineFromWorkspace returns the Unmarshaled fs.Pipeline struct from turbo.json in the given workspace.
+func (g *CompleteGraph) GetPipelineFromWorkspace(workspaceName string, isSinglePackage bool) (fs.Pipeline, error) {
+	turboConfig, err := g.GetTurboConfigFromWorkspace(workspaceName, isSinglePackage)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return turboConfig.Pipeline, nil
+}
+
+// GetTurboConfigFromWorkspace returns the Unmarshaled fs.TurboJSON from turbo.json in the given workspace.
+func (g *CompleteGraph) GetTurboConfigFromWorkspace(workspaceName string, isSinglePackage bool) (*fs.TurboJSON, error) {
+	cachedTurboConfig, ok := g.WorkspaceInfos.TurboConfigs[workspaceName]
+
+	if ok {
+		return cachedTurboConfig, nil
+	}
+
+	// Note: dir for the root workspace will be an empty string, and for
+	// other workspaces, it will be a relative path.
+	pkgJSON, err := g.GetPackageJSONFromWorkspace(workspaceName)
+	if err != nil {
+		return &fs.TurboJSON{}, nil
+	}
+
+	workspaceAbsolutePath := pkgJSON.Dir.RestoreAnchor(g.RepoRoot)
+	turboConfig, err := fs.LoadTurboConfig(workspaceAbsolutePath, pkgJSON, isSinglePackage)
+
+	// If we failed to load a TurboConfig, return the error
+	if err != nil {
```

Comment:
I kinda alluded to this in #3749 but you can distinguish between "not found" and "there was some error that needs to be handled". Without changing `fs.LoadTurboConfig`, you should be able to use `os.IsNotExist(err)`. It's also possible to change `fs.LoadTurboConfig` to handle that case separately. Really a judgement call of at what level do we explicitly check for "not found", but i think we need to split it out if we also want to return relevant errors to the user

Reply:
This was a bit tricky because `os.IsNotExist` will bubble is at the first invocation of `getTaskDefinition()`. This also throws an error when the task isn't found, and that is handled by ignoring the error:

https://github.com/vercel/turbo/blob/c4e39ed4b3b1e8101d6544a12196cc8c485ee7a0/cli/internal/core/engine.go#L140-L145

Took a stab at this by adding a new MissingTaskError in https://github.com/vercel/turbo/pull/3760. Let me know what you think